src/QMCDrivers/VMC/VMCBatched.cpp

```diff
@@ -247,6 +247,24 @@ void VMCBatched::process(xmlNodePtr node)
                                 qmcdriver_input_.get_walkers_per_rank(), 1.0, qmcdriver_input_.get_num_crowds());
 
     Base::startup(node, awc);
+
+
+    if (!qmcdriver_input_.get_variational_parameter_file().empty())
```

Comment:
This is not behavior that is meaningful to a VMCBatched driver not running inside of an optimizer, so this shouldn't be here. Move this code into the optimizer driver this will in turn make it easy to put the parameter into the right input class. 

It looks to me that the time to do it would be before constructing the vmcEngine so it is constructed with a "populated" golden wavefunction.
If you don't want that to be the one from the Wavefunction Pool, then make a clone of it.

I think this will also remove the need for set_variational_parameters to be called from this code block since your're not reading the parameters after you have cloned the gold wavefunction for the population.

Reply:
I don't think behavior should be controlled by the presense or absense of this file input. And the default behavior of a program should not be controlled by implicitly observed state of the working directory without a explicit input parameter or cli argument.

Reply:
So move the input into `<wavefunction/>` where the gold is constructed?

Reply:
> I don't think behavior should be controlled by the presense or absense of this file input. And the default behavior of a program should not be controlled by implicitly observed state of the working directory without a explicit input parameter or cli argument.

I have difficulty to follow this argument. You see an input tag and then act on it. It doesn't seem bothering "default".

Reply:
The input tag and reading of it should be handled by WfOptDriverInput.

This code block should be in QMCFixedSampleLinearOptimizeBatched::processOptXML not in VMCBatched.

Reply:
the variable name does not specify this is a file to load, just that it is a file name for the variational parameters.
The tag does say its to load but I had to drill into the implementation of the input reading to see that :( If I'm not coding the parsing of the input file I don't ever want to look at the text tags for input.

Where do the new variation parameters get saved and if I want to just use the same input.xml over and over 
again to keep optimizing (say in a premptable queue) can that be done?